### PR TITLE
Drop the cruft

### PR DESCRIPTION
**Pull request by cstamas**
Drop dependencies that are superfluous (one class used) or are deprecated:
* drop commons-collections4
* drop (to be deprecated) maven-dependency-tree

Slight updates and cleanup in dependencies.

---

https://issues.apache.org/jira/browse/MSHADE-473
https://issues.apache.org/jira/browse/MSHADE-474
https://issues.apache.org/jira/browse/MSHADE-475
https://issues.apache.org/jira/browse/MSHADE-476
https://issues.apache.org/jira/browse/MSHADE-477

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -185,21 +181,21 @@
       <artifactId>maven-dependency-tree</artifactId>
       <version>3.2.1</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
```
Comment (olamy) [inline pom.xml:185]: ? really needed? this doesn't look related to the subject of the PR

File: pom.xml [hunk 2]
```diff
@@ -185,21 +181,21 @@
...
+      <artifactId>commons-compress</artifactId>
+      <version>1.26.2</version>
+    </dependency>
     <dependency>
       <groupId>commons-io</groupId>
       <artifactId>commons-io</artifactId>
-      <version>2.13.0</version>
+      <version>2.16.1</version>
```
Comment (olamy) [inline pom.xml:192]: this doesn't look related to the subject of the PR
maybe change the subject

File: pom.xml [hunk 3]
```diff
@@ -79,12 +79,11 @@
   </distributionManagement>
 
   <properties>
-    <mavenVersion>3.6.3</mavenVersion>
+    <mavenVersion>3.9.7</mavenVersion>
```
Comment (olamy) [inline pom.xml:82]: this doesn't look related to the subject of the PR
maybe change the subject

File: pom.xml [hunk 4]
```diff
@@ -93,12 +92,12 @@
       <dependency>
         <groupId>org.eclipse.sisu</groupId>
         <artifactId>org.eclipse.sisu.inject</artifactId>
-        <version>${sisu.version}</version>
+        <version>${version.sisu-maven-plugin}</version>
```
Comment (olamy) [inline pom.xml:95]: the property name doesn't look related to such library.
I know it's coming from the parent but this is really confusing.

**Review by olamy — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (cstamas) [reply pom.xml:180]: ? unsure I get your comment. If you'd look carefully, you will see that I just moved this dependency from last place (as it is non test) to here. I just "moved" it to have proper dep order.

File: pom.xml (hunk 3 above)
Comment (cstamas) [reply pom.xml:82]: changed the subject

File: pom.xml (hunk 4 above)
Comment (cstamas) [reply pom.xml:95]: Yes, is ugly, will figure out something. My point was really to not have "disconnected" the sisu version in this plugin vs parent.

Comment (michael-o) [reply pom.xml:95]: Isn't this inherited anyway?

Comment (cstamas) [reply pom.xml:95]: Huh, seems it is, removing whole depMgt section and moving guice version to dep, and that's did it.

**Review by mosabua — APPROVED** [review]
Lgtm.

File: pom.xml (hunk 1 above)
Comment (olamy) [reply pom.xml:160]: ah right. sorry for that.
aren't we using spotless for pom?

File: pom.xml (hunk 2 above)
Comment (olamy) [reply pom.xml:167]: > changed the subject

thanks for that. The new subject is really more comprehensive :P

**Comment by olamy** [comment]
@cstamas, sorry, my bad. I merged it instead of only approving it,

**Comment by cstamas** [comment]
Np, this pr was "complete", so is ok to have it merged

**Comment by jira-importer** [comment]
Resolve #528

**Comment by jira-importer** [comment]
Resolve #720

**Comment by jira-importer** [comment]
Resolve #528

**Comment by jira-importer** [comment]
Resolve #719

**Comment by jira-importer** [comment]
Resolve #701

**Comment by jira-importer** [comment]
Resolve #720
